hypothesis-python/src/hypothesis/internal/conjecture/engine.py

```diff
@@ -616,10 +614,19 @@ def generate_new_examples(self):
             # to be whatever value is written there). That means that once we've
             # tried the zero value, there's nothing left for us to do, so we
             # exit early here.
-            for i in hrange(self.cap):
-                if i not in zero_data.forced_indices:
+            has_non_forced = False
+
+            # It's impossible to fall out of this loop normally because if we
+            # did then that would mean that all blocks are writes, so we would
+            # already have triggered the exhaustedness check on the tree and
+            # finished running.
+            for b in zero_data.blocks:  # pragma: no branch
```

Comment:
Minor: You could potentially get rid of this pragma, and put `no cover` on a separate `else` branch instead (with or without an `assert False`).

Reply:
I think that's logically true of *every* `no branch` pragma isn't it? Is there a particular reason you'd favour that here?

Reply:
I guess it only matters if you actually want to assert that the fall-off never happens.

Either way I don't feel strongly about it.

Reply:
I'm comfortable not adding the assertion. I think the worst case scenario here is that we do some extra work we didn't need to, and I'm pretty confident in the reasoning that it's impossible.